JSqlServerBulkInsert/src/main/java/de/bytefish/jsqlserverbulkinsert/converters/BaseConverter.java

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Philipp Wagner and Victor Lee. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package de.bytefish.jsqlserverbulkinsert.converters;
+
+public abstract class BaseConverter<TSourceType> implements IConverter<TSourceType> {
+
+    public Object convert(TSourceType value) {
+        if(value == null) {
+            return null;
+        }
+
+        return internalConvert(value);
+    }
+
+    public abstract Object internalConvert(TSourceType value);
```

Comment:
If we are looking for items to debate, one would be whether to have this be abstract or whether it should have a default implementation that is overridden only when needed. Since most of the overrides simply `return value;`, that might be considered as the default implementation. Then we would avoid a lot of duplicate code. On the other hand, keeping it abstract as you have it now, forces the implementer to think specifically about what they are doing. And you have already written all of the code anyways so the effort is done.

Reply:
Yes, I think we should make a common class for all versions, that just return the value. I still would like to have the base class being abstract as it is. What do you think could a converter be called, that just returns a value? IdentityConverter? 🤔

Reply:
Not a lot of synonyms for 'no change' or 'no conversion'. So IdentityConverter is fine.

Reply:
Yes, you were right. Removing those duplicated code boiled down the number of converters a lot and makes the code much easier to understand.